esbmc_ai/esbmc_util.py

```diff
@@ -94,10 +98,187 @@ def esbmc(path: str, esbmc_params: list, timeout: Optional[float] = None):
+    esbmc_output = re.sub(pattern, '', esbmc_output, flags=re.MULTILINE)
+    
+    esbmc_output = re.sub(r'\n\s*\n', '\n', esbmc_output)
+    esbmc_output = esbmc_output.replace("output from ESBMC", "ESBMC output")
+    esbmc_output = re.sub(r'The time.*', '', esbmc_output, flags=re.MULTILINE) 
+    
+    esbmc_output = re.sub(r'[^.]VCCs represent.*|[^.]These VCCs represent.*', '', esbmc_output, flags=re.MULTILINE) #These types of lines represent some sequences of VCCs that can easily be ommitted for development purposes
+    esbmc_output = re.sub(r'.*bitvector.*', '', esbmc_output, flags=re.MULTILINE)
+    esbmc_output = re.sub(r'\s*\(\s*`thread\d+`(?:,\s*`thread\d+`)*\s*\)', '', esbmc_output)
+    
+    esbmc_output = re.sub(r'.*floating-point arithmetic.*', '', esbmc_output, flags=re.MULTILINE)
+    esbmc_output = re.sub(r'.*encoding.*', '', esbmc_output, flags = re.MULTILINE)
+    esbmc_output = re.sub(r'.*encoding time.*', '', esbmc_output, flags= re.MULTILINE)
+
+    esbmc_output = re.sub(r', which takes \d+\.\d+s\.*', '.', esbmc_output, flags=re.MULTILINE)
+    esbmc_output = re.sub(r'The output indicates that ESBMC generates a verification condition \(VCC\) for each possible path and encodes them into a solver\.', '', esbmc_output, flags= re.MULTILINE) #This information can be disregarded for development purposes
+    esbmc_output = re.sub(r'Measures the [^.]+?encoding time [^.]+?\.', '', esbmc_output, flags=re.MULTILINE)
+
+    esbmc_output = re.sub(r'In this case, the procedure took [0-9.]+ seconds\.', '', esbmc_output, flags=re.MULTILINE)
+    esbmc_output = re.sub(r'\s*It measures[^.]+? encoding time [^.]+?\.', '', esbmc_output, flags=re.MULTILINE)
+    esbmc_output = re.sub(r'[^.]*GOTO program[^.]*\.', '', esbmc_output)
+
+    esbmc_output = re.sub(r'\d+\.\d+s \.', '', esbmc_output, flags=re.MULTILINE)
+    esbmc_output = re.sub(r'\.\s*.*time.*\n\n|\.\s*[^.]*time[^.]*\.', '', esbmc_output, flags=re.MULTILINE)    
+    esbmc_output =  re.sub(r'.*program time: \d+\.\d+s \.', '',esbmc_output, flags=re.MULTILINE)
+
+    esbmc_output = re.sub(r'\.\s*[^.]*?BMC program time:\d+\.\d+s', '', esbmc_output, flags= re.MULTILINE)
+    esbmc_output =re.sub(r'\.\s*[^.]*?Runtime decision procedure: \d+.\d+s\"*', '', esbmc_output, flags=re.MULTILINE)
+    esbmc_output = re.sub(r'.*Symex.*|.*symex.*|.*symbolic execution.*\n\n', '', esbmc_output, flags=re.MULTILINE)
+
+    esbmc_output =re.sub(r'After[^.]*?, the code', 'The program then', esbmc_output, flags=re.MULTILINE)
+    esbmc_output = re.sub(r'.*assignments.*|.*assignment.*', '', esbmc_output, flags=re.MULTILINE)
+    esbmc_output = re.sub(r'.*iteratively.*', '', esbmc_output, flags=re.MULTILINE)
+
+    esbmc_output = re.sub(r'.*branches.*', '', esbmc_output, flags=re.MULTILINE) #The informations about branches are not necessary relevant to the understanding of bugs
```

Comment:
This regex `.*branches.*` removes anything with the word branch. Consider the following code:

```c
void main() {
	char branches = 3;
}
```

This would turn to:

```c
void main() {
}
```

There are a lot more regex that are so general that they would filter out a lot of code... Too much to point to.